vnext/docs/E2ETest.md

```diff
@@ -0,0 +1,430 @@
+# Author and Run E2E Test for React Native Windows
+
+# Overview
+
+## Technical Decisions
+
+### MSTest vs node test runner
```

Comment:
I dont think we need all the discussion about why this runner and why this driver etc in the repo.
What we need is how do I run the tests, how do I write the tests, how do I debug the tests etc.

Reply:
reasonable. I will split them to two files.
Somebody like me may still be interested in this stuff.